docusaurus/docs/create-a-plugin/develop-a-plugin/best-practices.md

```diff
@@ -0,0 +1,64 @@
+---
+id: best-practices
+title: Best practices
+description: An index of best practices for plugins.
+keywords:
+  - grafana
+  - plugins
+  - plugin
+  - best practices
+sidebar_position: 5
+---
+
+This document summarizes best practices for Grafana plugins. This live document is updated as new best practices are identified.
+
+## General
+
+- Choose the right plugin type for your use case: panel, data-source or app. See [Plugin types](/introduction/plugin-types-usage) for more information.
+- Verify that your data-source or app plugin can be provisioned. See [Provisioning](https://grafana.com/docs/grafana/latest/administration/provisioning/#data-sources) for more information.
+- Include default dashboards in your data-source or app plugin. See [Bundling of dashboards](https://grafana.com/docs/grafana/latest/administration/provisioning/#data-sources) for more information.
+- Ensure that the minimum version of Grafana is correct. This should point to the earliest version of Grafana that your plugin fully supports. This is specified as `grafanaDependency` in your plugin.json.
+
+## Panel plugins
+
+- Do not store or use credentials. Panel plugins do not have a way to securely store credentials. If your plugin needs to use credentials, consider using a data-source plugin instead and a panel plugin to display the information returned by the data-source.
+- See how you can add more advanced options to your panel plugin [here](/create-a-plugin/extend-a-plugin/custom-panel-option-editors).
```

Comment:
This feels like it depends on the panel rather than being a best practice which could be generally applied by developers?

Reply:
I thought this to be a bit of feature discovery as well. Maybe someone is doing something cumbersome with the pre-defined panel options because they don't know you can do custom ones. WDYT?

Reply:
That's fair. How about something like "If the default panel options are not a good fit for what you're trying to offer users, consider creating custom options"